rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,181 @@
+  explanation is likely as bad as commits without any description, rationale,
+  or explanation in the body.
+
+For minor deviations from these guidelines, the community will normally favor
+reminding the contributor of this policy over reverting or blocking a commmit /
+PR.
+
+Commits and PRs without a title and/or a body are not considered minor
+deviations from these guidelines and hence must be avoided.
+
+Most importantly, the contents of the commit message, especially the body,
+should be written to convey the intention of the change, so it should avoid
+being vague. For example, commits with a title like “Fix”, “Cleanup”, and
+“Fix flaky test” and without any body text should be avoided. Also, for the
+review, it will leave the reviewer wondering about what exactly was fixed or
+changed and why the change is necessary, slowing the review.
+
+Below is an example that can be used as a model:
+
+> [microTVM] Zephyr: Remove zephyr_board option from build, flash, and open_transport methods
+>
+> Currently it’s necessary to pass the board type via ‘zephyr_board’ option to
+> the Project API build, flash, and open_transport methods.
+>
+> However, since the board type is already configured when the project is
+> created (i.e. when the generate_project method is called), it’s possible to
+> avoid this redundancy by obtaining the board type from the project
+> configuration files.
+>
+> This commit adds code to obtain the board type from the project CMake files,
+> removing this option from build, flash, and open_transport methods, so it’s
+> only necessary to specify the ‘zephyr_board’ option when calling
+> generate_project.
+>
+> This commit also moves the ‘verbose’ and ‘west_cmd’ options from ‘build’
+> method to ‘generate_project’, reducing further the number of required options
+> when building a project, since the ‘build’ method is usually called more often
+> than the ‘generate_project’.
+
+If a PR is created from more than one commit (a patchset), then ideally the
+changes should be split into commits in a reasonable way. For instance, if it’s
+a fix and so it also adds a regression test, it’s a good practice to split the
+changes having one commit for the fix and another one for the test, applying the
+rules and recommendations in this guideline to each commit.
+
+After a new PR is created and the review starts it’s common that reviewers will
+request changes. Usually the author will address the reviewers’ comments and
+push additional commits on top of the initial ones. For these additional commits
+there is no recommendation regarding the commit messages. The only
+recommendation is that if the additional commits render the PR title and/or body
+outdated then the PR title and body must reflect the new changes in the code and
+be updated accordingly (remember that the PR title and body will be used to
+compose the final commit message that will land in the main tree).
```

Comment:
Committers have the ability to edit others' messages, I believe, so would it be incumbent on the committer who is merging the PR to ensure the topic post is up-to-date?

Reply:
I guess it would be easier for the committers to request updates from the author before merging

Reply:
I'd say ideally we want it to be the author's responsibility, otherwise this will end up always been left to the committer. That being said, the committer should always seek to fix any issues prior to committing but they should retain the right to inform the author of the rules and encourage them to follow them in future. We probably need more explicit language around the author/committer responsibilities.

Reply:
I agree with @mbaret I'll improve this part and post a v3.

Reply:
Addressed here: https://github.com/apache/tvm-rfcs/pull/88/commits/afcbb981155b5a7eaf99c8306ee1cd2fb33aa463

Reply:
@mbaret can you have a look and mark it as resolved In case you agree, or propose a suggestion in case this needs further improvement?

Reply:
seems good - will leave to @slyubomirsky to mark as resolved though given it's his chain.